src/metadata.jl

```diff
@@ -0,0 +1,37 @@
+function _versioncheck(version::VersionNumber)
+    supported = first(VALID_VERSIONS) <= version < last(VALID_VERSIONS)
+    supported || error(LOGGER, ArgumentError(
+        string(
+            "Unsupported version ($version). ",
+            "Expected a value between ($VALID_VERSIONS)."
+        )
+    ))
+end
+
+
+# Cache of the versioninfo and image, so we don't compute these every time.
+const _CACHE = Dict(
```

Comment:
I think it is fine to move this `const _CACHE` here
as it isn't really a package configuration constant kinda deal,
it is a implementation detail of `_pkgs` and of `_image`.